src/wing/CTiglWingConnection.cpp

```diff
@@ -75,53 +77,44 @@ int CTiglWingConnection::GetSectionElementIndex() const
 // Returns the wing profile referenced by this connection
 CCPACSWingProfile& CTiglWingConnection::GetProfile()
 {
-    return segment->GetWing().GetConfiguration().GetWingProfile(*m_resolved->profileUIDPtr);
+    return segment->GetUIDManager().ResolveObject<CCPACSWingProfile>(*m_resolved->profileUIDPtr);
 }
 
 const CCPACSWingProfile& CTiglWingConnection::GetProfile() const
 {
-    return segment->GetWing().GetConfiguration().GetWingProfile(*m_resolved->profileUIDPtr);
+    return segment->GetUIDManager().ResolveObject<CCPACSWingProfile>(*m_resolved->profileUIDPtr);
 }
 
 // Returns the positioning transformation (segment transformation) for the referenced section
 CTiglTransformation CTiglWingConnection::GetPositioningTransformation() const
 {
-    return segment->GetWing().GetPositioningTransformation(GetSectionUID());
+    // TODO: this is a bit ugly
```

Comment:
@bernhardmgruber Do you have a good idea, how to improve this code? Boost variant could be an option, but I don't like to have boost::variant in the public interface.

Reply:
there was a trade-off when I designed the parent getter. I thought about variant as it is the correct choice when having a multitude of types with a single active one. However, i feared you would not accept the PR, as the interface of variant is not trivial for novices. That is why I designed the interface to be used by if statements just as the ones you wrote.

So, apart from putting segment->GetParent() into a local reference to safe some chars, this is the best you can do.

Reply:
Okay... then I'll let this as it is. If an if is ommited here, we should get an exception and therefore we will notice this.

Thank you!